com.regnosys.rosetta/src/com/regnosys/rosetta/RosettaExtensions.xtend

```diff
@@ -221,4 +230,21 @@ class RosettaExtensions {
 		}
 		return false
 	}
+	
+	def boolean isChoiceRuleCondition(Condition cond) {
+		return cond.constraint !== null
```

Comment:
Something doesn't feel right about this method. We are either testing for existence of Constraints, in which case the name of the function should reflect that; or we are checking for choice rules, in which case the predicate needs to update?

Reply:
It describes the difference between Data rules and Choice rules. That is something you need to consolidate as one validation-concept; over all layers.

Reply:
But `cond.constraint !== null` includes Choice rules as well as OneOfs, so we are treating those 2 things as the same thing in this context. Let's take the naming conversation offline and we can address in subsequence PR.